libpod/container_internal.go

```diff
@@ -1360,19 +1360,57 @@ func (c *Container) addImageVolumes(ctx context.Context, g *generate.Generator)
 			continue
 		}
 		volumePath := filepath.Join(c.config.StaticDir, "volumes", k)
-		if _, err := os.Stat(volumePath); os.IsNotExist(err) {
+		srcPath := filepath.Join(mountPoint, k)
+
+		if _, err := os.Stat(srcPath); os.IsNotExist(err) {
+			logrus.Infof("Volume image mount point %s does not exits in root FS, need create it", k)
 			if err = os.MkdirAll(volumePath, 0755); err != nil {
 				return errors.Wrapf(err, "error creating directory %q for volume %q in container %q", volumePath, k, c.ID)
 			}
+
+			if c.config.User != "" {
+				if !c.state.Mounted {
+					return errors.Wrapf(ErrCtrStateInvalid, "container %s must be mounted in order to translate User field", c.ID())
+				}
+				uid, gid, err := chrootuser.GetUser(c.state.Mountpoint, c.config.User)
+				if err != nil {
+					return err
+				}
+
+				if err = os.Chown(volumePath, int(uid), int(gid)); err != nil {
```

Comment:
I don't think this is the logic we want to use - we should use the owner and permissions of the parent folder in the image (parent path of `filepath.Join(mountPoint, k)`)

Reply:
I agree that the way I did may not be optimal. 

But I don't think that using the parent folder stats is correct, because it's likely that the parent folder has stricter permissions. E.g. when mounting a volume as sub folder of ```/mnt```, if we use the same permissions of ```/mnt``` (root - 755), we end up again with a non-accessible folder for the container user.

Reply:
@rhatdan, my implementation mimics Docker in this regard:
```
# ./bin/podman run --rm volumetest ls -lad /data      
drwxr-xr-x    1 3267     3267             0 Jun 18 16:29 /data
# ./bin/podman run --user 1234 --rm volumetest ls -lad /data
drwxr-xr-x    1 3267     3267             0 Jun 18 16:29 /data
```

Reply:
@mheon Does that look good to you?